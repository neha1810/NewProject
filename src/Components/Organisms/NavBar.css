/* .navBar{
    position: sticky;
    top:0;
    background: whitesmoke;
    z-index: 1000;
    margin-right: 1%;
    margin-left: 1%;
   } */