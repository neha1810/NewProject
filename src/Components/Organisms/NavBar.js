import React from 'react';
import '../../../node_modules/bootstrap/dist/css/bootstrap.min.css'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { fetchProductData } from '../Redux/actions/productAction'
import { historyDetails } from '../Redux/actions/historyAction'


const NavItem = props => {
  const pageURI = window.location.pathname + window.location.search
  const liClassName = (props.path === pageURI) ? "nav-item active" : "nav-item";
  const aClassName = props.disabled ? "nav-link disabled" : "nav-link"
  return (
    <li className={liClassName}>
      <a href={props.path} className={aClassName}>
        {props.name}
        {(props.path === pageURI) ? (<span className="sr-only">(current)</span>) : ''}
      </a>
    </li>
  );
}



class navBar extends React.Component {
  linkClicked = event => {


    this.props.addLink(event.target.name)
  }
  linkHistoryClicked = event => {


    this.props.historyLink(event.target.name)
  }
  render() {
    return (
      <nav className="navbar navbar-expand-lg ">
        {/* <a className="navbar-brand " href="/"></a> */}
        <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon"></span>
        </button>

        <div className="collapse navbar-collapse " id="navbarSupportedContent">
          <ul className="navbar-nav mr-auto ">
            <Link to="/products" name="ecommerce" onClick={this.linkClicked}  className="myNav" >Ecommerce</Link>
            <Link to="/entertainment" name="Entertainment" onClick={this.linkClicked}  className="myNav" >Entertainment</Link>
            {/* <NavItem path="/" name="Ecommerce" />
            <NavItem path="/page2" name="Entertainment" />
            <NavItem path="/page3" name="Travel and hospitality"  />
            <NavItem path="/page3" name="Health and beauty"  />
            <NavItem path="/page3" name="Food and beverages"  /> */}

            <Link to="/History" name="history" onClick={this.linkHistoryClicked} className="myNav" >History</Link>
          </ul>

        </div>
      </nav>
    )
  }
}
const mapDispatchToProps = dispatch => {
  return {


    addLink: event => dispatch(fetchProductData(event)),
    historyLink: event => dispatch(historyDetails(event))


  }
}
export default connect(
  null,
  mapDispatchToProps
)(navBar);

