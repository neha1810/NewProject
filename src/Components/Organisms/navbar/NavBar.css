/* .navBar{
    position: sticky;
    top:0;
    background: whitesmoke;
    z-index: 1000;
    margin-right: 1%;
    margin-left: 1%;
   } */
   /* .navBar
   {
    position: fixed;
    width: 100%;
    top: 0;
   } */