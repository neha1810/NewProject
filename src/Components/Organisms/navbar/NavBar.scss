.navBar{
    position: sticky;
    top:0;
    background: whitesmoke;
    z-index: 1000;
    margin-right: 1%;
    margin-left: 1%;

   }
 #navBar
   {
    background: #16A085;
    margin:0
   } 
   p{
    margin: 0%;
   }
   .myNav
   {
    color:white
   }
   #navbarDropdown
   {
       color:white
   }
   #logo
   {
       width:250px;
       height:50px
   }
   #mybanner
   {
    height: 50vh;
    width: 100%;

   }
   #head
   {
       padding: 10px
   }
   #cart
   {
          padding-right: 40px;
   }