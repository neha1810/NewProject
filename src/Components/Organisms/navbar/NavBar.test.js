import React from 'react';
import NavBar from './NavBar';
import {shallow} from 'enzyme';
// setup file
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('NavBar', () => {
let wrapper;
beforeEach(()=> {
wrapper = shallow(
        <NavBar />
    )
})


it('render the NavBar component', () => {
    expect(wrapper).toMatchSnapshot();
});
it('Button exists',()=> {
    //wrapper.find('button').at(1).simulate('click');
    expect(wrapper.find('#NavBar')).toBeDefined();
})



})