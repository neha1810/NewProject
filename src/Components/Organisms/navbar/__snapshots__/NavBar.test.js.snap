// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar render the NavBar component 1`] = `undefined`;
